R/calc_mutation_frequency_bin_region.R

```diff
@@ -0,0 +1,294 @@
+                                               window_size = 1000,
+                                               return_format = "long",
+                                               min_count_per_bin = 0,
+                                               return_count = TRUE,
+                                               drop_unmutated = FALSE,
+                                               ...) {
+  
+  #check if any invalid parameters are provided
+  check_excess_params(...)
+  
+  # Create objects to describe region both as string and individual objects
+  try(if (missing(region) & missing(chromosome)) {
+    stop("No region information provided. Please provide a region as a string in the chrom:start-end format, or as individual arguments. ")
+  })
+  
+  if ((drop_unmutated | min_count_per_bin > 0) & return_format == "wide") {
+    message("To return a wide table, all samples and windows must be kept. Ignoring drop_unmutated and min_count_per_bin arguments. ")
+  }
+  
+  if (missing(region)) {
+    region <- paste0(
+      chromosome, ":", start_pos, "-",
+      end_pos
+    )
+  } else {
+    chunks <- region_to_chunks(region)
+    chromosome <- chunks$chromosome
+    start_pos <- as.numeric(chunks$start)
+    end_pos <- as.numeric(chunks$end)
+  }
+  
+  # Harmonize metadata and sample IDs
+  metadata <- id_ease(
+    these_samples_metadata,
+    these_sample_ids,
+    this_seq_type
+  )
+  these_sample_ids <- metadata$sample_id
+  
+  
+  if (
+    (str_detect(chromosome, "chr") & projection == "grch37") |
+    (!str_detect(chromosome, "chr") & projection == "hg38")
+  ) {
+    stop("chr prefixing status of region and specified projection don't match. ")
+  }
+  
+  
+  # Check region size and compare to max region size
+  # Is this really needed?
+  max_region <- 5e+06
+  
+  region_size <- end_pos - start_pos
+  if (region_size < max_region) {
+    message(paste(
+      "processing bins of size", window_size,
+      "across", region_size, "bp region"
+    ))
+  } else {
+    message(paste("CAUTION!\n", region_size, "exceeds maximum size recommended by this function."))
+  }
+  
+  # Split region into windows
+  windows <- data.frame(
+    chrom = chromosome,
+    window_start = seq(start_pos, end_pos, by = slide_by)
+  ) %>%
+    dplyr::mutate(window_end = window_start + window_size - 1) %>%
+    dplyr::select(chrom, window_start, window_end)
+  
+  # Option to return full region count instead of sliding window
+  if (window_size == 0) {
+    windows <- data.frame(
+      chrom = chromosome,
+      window_start = start_pos,
+      window_end = end_pos
+    )
+  }
+  
+  # Obtain SSM coordinates from GAMBL if no maf_data was provided
+  if (is.null(maf_data)) {
+    try(
+      if (!"seq_type" %in% colnames(metadata)) {
+        stop("seq_type must be present in metdata for compatibility with get_ssm_by_sample")
+      }
+    )
+    message("Using GAMBLR.results::get_ssm_by_region...")
```

Comment:
I missed this initially. This should really read `Using the GAMBLR.data::get_ssm_by_region...`, right?

Reply:
Oh, yes 100% Thanks!